account_payment_purchase/models/account_invoice.py

```diff
@@ -10,9 +10,13 @@ class AccountMove(models.Model):
 
     @api.onchange("purchase_vendor_bill_id", "purchase_id")
     def _onchange_purchase_auto_complete(self):
+        new_purchase = self.purchase_id
         new_mode = self.purchase_id.payment_mode_id.id or False
         new_bank = self.purchase_id.supplier_partner_bank_id.id or False
         res = super()._onchange_purchase_auto_complete() or {}
+        if not new_purchase:
```

Comment:
Why not directly?:
```suggestion
        if not self.purchase_id:
```

Reply:
because the _onchange_purchase_auto_complete method sets self.purchase_id to False

Reply:
And do it before at the beginning

Reply:
OK, and can't be done at the beginning, returning super in that case?

Reply:
Yes, I tested and I can do it before the onchange. Just I moment and I will push the changes.